### ISSUE TITLE
Partial and Pick do not compose as expected.

### ISSUE THREAD

**Issue by wXAtxS5wQcJKntslStMQKflS5** [microsoft/TypeScript#46210]
# Bug Report

### 🔎 Search Terms

Partial, pick, compose. 

### 🕗 Version & Regression Information

- This is a crash: no
- This is the behavior in every version I tried, and I reviewed the FAQ for entries about partial, pick.

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=4.5.0-dev.20211004&ssl=5&ssc=1&pln=6&pc=1#code/C4TwDgpgBKkAwEYoF4oG8CwAoAkAOwEMBbCALigGdgAnASzwHNscCGyo8BXIgIwmuwBfANzZssaBLgAmFOmaES5KnUbMA9gGNNnMAWC11eZTXpMsgsVgBmnPJoNGYEKogAUANwIAbTuwAKBNQGPgA8UggAfACU6FCaRhTq3hAAdN7qDJ4+fqmKEAA0UF6+aawQ0cJQlljYCXhUxTkQiFDkUrKoaPnkAOQAwgAWEMBQAGIEANb8vUVaOnqOxlC9RJwUtJq0BHi9NQD0+86uCNmlFIixh1AAIuouHOqjCURg3rRpVvWNJX4y5P5NpNwuAWtIir18r0oAAfFbzXT6Qy7SJyX4tBCiLDAFzAdzomSVKDXACSABMID5vCAoAB3dScbxkx7PdSvWgpVJAA)

### 💻 Code

```ts
type type01 = {
	name: string
	age: number
};

type type02 = {
	name: string
	occupation: string
}

function test01(value: Partial<type01>) { console.log(value.name, value.age); }

const value01 : type02 = {name: 'Chet Faker', occupation: 'musician'}
// test01(values01) // Does not complie.

const value02: Pick<type02, 'name' | 'occupation'> = value01;
test01(value02); // Ideally would not compile.
```

### 🙁 Actual behavior

Compiled without complaint.

### 🙂 Expected behavior

Would not compile.

**Comment by MartinJohns** [comment]
Why do you think it should not compile? Looks perfectly fine and as expected.

**Comment by wXAtxS5wQcJKntslStMQKflS5** [comment]
@MartinJohns 
```
Partial<Type>
Constructs a type with all properties of Type set to optional. This utility will return a type that represents all subsets of a given type.
```
```
Pick<Type, Keys>
Constructs a type by picking the set of properties Keys (string literal or union of string literals) from Type.
```
Both `Pick<type02, 'name' | 'occupation'>` and `type02` are not a subset of `type01`. https://www.typescriptlang.org/docs/handbook/utility-types.html

**Comment by fatcerberus** [comment]
This is correct.  `Partial` makes all properties optional, and extra properties are allowed in general.  Thus `value01` is correctly assignable to `Partial<type01>`.

**Comment by wXAtxS5wQcJKntslStMQKflS5** [comment]
@fatcerberus Yep. Ideally,
> and extra properties are allowed in general

Would be called out within https://www.typescriptlang.org/docs/handbook/utility-types.html. This behavior caught me off guard.

Blue skying a bit here. Wouldn't a `Subset` utility type be nice?

```ts
type Person = {
    first: string, last: string
}

type Subset<T, Shape> =
  Exclude<keyof T, keyof Shape> extends never ? 
  T : never;

declare function savePerson<T>(person: Subset<T, Person>): void;

const tooFew = { first: 'Stefan' };
const exact = { first: 'Stefan', last: 'Baumgartner' }
const tooMany = { first: 'Stefan', last: 'Baumgartner', age: 37 }

savePerson(tooFew); // ✅ doesn't work
savePerson(exact); // ✅ satisfies the contract
savePerson(tooMany); // 💥 doesn't work
```
This block of code is a based on https://fettblog.eu/typescript-match-the-exact-object-shape/.

**Comment by fatcerberus** [comment]
The fact that extra properties can be present is a basic fact of the type system ("structural subtyping" is the term used IIRC) and not something specific to `Partial`.  What it seems you might be looking for is Exact Types: #12936
